### PR TITLE
Add methodology evidence study

### PR DESCRIPTION
**Pull request by graphanov**
## Summary
- Add `osc study [--json] [--since] [--out]` for honesty-bounded methodology evidence derived from local git and `.osc/` artifacts.
- Add the methodology protocol, first self-study, closed plan 125 evidence note, and the follow-up plan 126 backlog context.
- Add tests for the study schema, source labels, null handling, no-imputation guards, markdown rendering, and CLI output.

## Verification
- `npm run osc -- plan validate .osc/plans/done/125-methodology-evidence-harness.md`
- `./verify.sh --strict`
- `npm test -- --run tests/study.test.ts`
- `npm run osc -- study --json --out /tmp/hermes-study-125-final.json` + JSON source checks
- `npm run osc -- study --out /tmp/hermes-self-study-125-final.md` + timestamp-normalized comparison against `docs/EVIDENCE_SELF_STUDY.md`
- `npm run build`
- `npm test -- --run`
- `git diff --cached --check` before commit

## Risk / gates
- `src/study.ts` is allowlisted in the runtime-omx no-spawn boundary test as a local-git reader, matching the existing `metrics.ts` pattern. It reads local git only and does not spawn runtimes.
- The self-study intentionally reports null/negative findings; it does not make causal, benchmark, telemetry, provider, model, or compliance claims.
- A/B comparison remains protocol-only text and is not run in this PR.
- No npm publish, GitHub Release, merge, or public value claim is included in this PR.

### CODE REVIEW

**Comment by graphanov** [comment]
@codex review

File: src/study.ts [hunk 1]
```diff
@@ -0,0 +1,471 @@
...
+    to: shas.length > 0 ? shortSha(shas[0]) : null,
+    commit_count: shas.length,
+    source: 'git',
+  };
+}
+
+function listInScopePlanSlugs(root: string, planDetails: Array<{ slug: string; path: string }>): string[] {
+  // Restrict to the plans the metrics layer kept in scope (respects --since); fall back to disk scan.
+  if (planDetails.length > 0) return planDetails.map((plan) => plan.slug);
+  const slugs = new Set<string>();
```
[the remark this answers is not in the document]
Comment (graphanov) [reply src/study.ts]: Fixed in 6722f60.

Change: `osc study --since ...` now scopes plan-reference commit counting to the exact metrics-filtered plan set. If no plans are in scope, historical plan slugs do not leak back in from disk and the count is `0`.

Regression: added `does not count historical plan slugs when --since leaves no plans in scope` in `tests/study.test.ts`.

Verification:
- `npm test -- --run tests/study.test.ts` — PASS, 15/15
- `npm run build` — PASS
- `npm test -- --run` — PASS, 441/441
- `./verify.sh --strict` — PASS, 9 pass / 0 fail / 1 existing worktree warning
- `git diff --check` — PASS

File: MISSION.md [hunk 2]
```diff
@@ -29,6 +29,7 @@ Explicit things this project is NOT trying to do. Legitimate scope discipline st
 One-line dated entries for every scope pivot. Format: `YYYY-MM-DD: <one-line pivot description + link to amendment file if applicable>`. Append entries in chronological order. Never rewrite history here.
 
 <!-- append YYYY-MM-DD entries below this line -->
+- 2026-05-29: closed 125-methodology-evidence-harness — Close methodology evidence harness
```
[the remark this answers is not in the document]
Comment (graphanov) [reply MISSION.md]: Fixed in 6722f60.

Change: corrected the `MISSION.md` changelog entry from `2026-05-29` to `2026-05-28` so it matches the actual evidence/commit chronology.

Verification:
- `npm test -- --run tests/study.test.ts` — PASS, 15/15
- `npm run build` — PASS
- `npm test -- --run` — PASS, 441/441
- `./verify.sh --strict` — PASS, 9 pass / 0 fail / 1 existing worktree warning
- `git diff --check` — PASS

**Comment by graphanov** [comment]
@codex review

File: docs/EVIDENCE_SELF_STUDY.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
+# Open Scaffold methodology evidence — self-study
+
+Computed at: 2026-05-28T22:22:35.736Z
+Commit range: 8be590907210..130dd082c736 (207 commits)
+Since filter: none (full history)
```
[the remark this answers is not in the document]
Comment (graphanov) [reply docs/EVIDENCE_SELF_STUDY.md:5]: Fixed in 9b3a761.

Change: markdown self-study output now labels the current endpoint as `HEAD` while the JSON report still carries the exact SHA. The self-study was regenerated after creating the final local commit and amended before push, so regenerating it at the reviewed commit now matches after normalizing only `Computed at:`.

Verification:
- `npm run osc -- study --out /tmp/hermes-self-study-after-amend.md` + timestamp-normalized compare — PASS
- `npm test -- --run tests/study.test.ts` — PASS, 15/15
- `npm run build` — PASS
- `npm test -- --run` — PASS, 441/441 before the self-study-only amend
- `./verify.sh --strict` — PASS, 9 pass / 0 fail / 1 existing worktree warning
- `git diff --check` — PASS

File: docs/AB_COMPARISON_PROTOCOL.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+## Held constant (controls)
+
+Same agent/model and tool access across both arms; same per-task time budget; same definition of done; same task-difficulty distribution; same environment; the same blinded reviewer and the same scoring rubric. The single intended difference between arms is the presence of the Open Scaffold loop.
+
+## Metrics
+
+Every figure is source-labeled. Where a metric is computable by `osc study` (plan `125`), reuse that computation so the observational and controlled views stay consistent; the remaining metrics are coded manually under a fixed rubric.
+
+- **Resume time (H1).** Wall-clock from the start of a session on a previously-paused task to the first commit or edit that advances the task. "Cold" is defined precisely: at least a 24-hour gap, a fresh session, and no access to prior scrollback. (`osc study` can approximate this observationally from commit timestamps; the controlled run measures it directly.)
```
[the remark this answers is not in the document]
Comment (graphanov) [reply docs/AB_COMPARISON_PROTOCOL.md]: Fixed in 9b3a761.

Change: removed the statement that `osc study` can approximate resume time from commit timestamps. The A/B protocol now says `osc study` deliberately reports resume time as unavailable in the observational self-study, and the controlled run must measure it directly.

Verification:
- `npm test -- --run tests/study.test.ts` — PASS, 15/15
- `npm run build` — PASS
- `npm test -- --run` — PASS, 441/441 before the self-study-only amend
- `./verify.sh --strict` — PASS, 9 pass / 0 fail / 1 existing worktree warning
- `git diff --check` — PASS

**Comment by graphanov** [comment]
@codex review

File: src/study.ts [hunk 5]
```diff
@@ -0,0 +1,460 @@
...
+  const since = options.since ?? null;
+  const hasGit = gitAvailable(root);
+
+  const metrics = computeMetrics({ root, now, since: since ?? undefined });
+  const planDetails = metrics.plans.map((plan) => ({ slug: plan.slug, path: plan.path }));
+  const amendments = amendmentCounts(root, planDetails);
+  const commitRange = computeCommitRange(root, hasGit, since ?? undefined);
+  const slugs = listInScopePlanSlugs(planDetails);
+  const planRefCommits = hasGit ? countPlanReferencingCommits(root, slugs, since ?? undefined) : null;
+  const gitSource: SignalSource = hasGit ? 'git' : 'unavailable';
```
[the remark this answers is not in the document]
Comment (graphanov) [reply src/study.ts]: Fixed in dc3c1fd.

Change: git-derived signal sources now come from `commitRange.source`, not from `rev-parse` alone. If `git log` cannot produce a commit range, both `commits_in_range` and `plan_referencing_commits` are `source: unavailable` with `value: null`.

Regression: added `marks git-derived signals unavailable when git log cannot read a commit range` for a freshly initialized repo with no commits.

Verification:
- `npm test -- --run tests/study.test.ts` — PASS, 16/16
- `npm run build` — PASS
- `npm test -- --run` — PASS, 442/442 before the self-study-only amend
- `npm run osc -- study --out /tmp/hermes-self-study-git-source-amend.md` + timestamp-normalized compare — PASS
- `./verify.sh --strict` — PASS, 9 pass / 0 fail / 1 existing worktree warning
- `git diff --check` — PASS

**Comment by graphanov** [comment]
@codex review

File: src/study.ts [hunk 6]
```diff
@@ -0,0 +1,460 @@
...
+      `Traceability floor: only ${refCommits.value} commit subject(s) name a known plan slug. This is a floor, not a total — commits that omit the slug are uncounted, so the figure understates real plan-to-commit traceability.`,
+    );
+  }
+  return headlines;
+}
+
+export function renderStudyMarkdown(report: StudyReport): string {
+  const range = report.commit_range;
+  const rangeText = range.source === 'git' && range.from && range.to
+    ? `${range.from}..HEAD (${range.commit_count ?? 0} commit${range.commit_count === 1 ? '' : 's'})`
```
[the remark this answers is not in the document]
Comment (graphanov) [reply src/study.ts]: Fixed in 9bd68c4.

Change: the self-study markdown no longer renders the span as Git `from..HEAD` syntax. It now says `<oldest> through HEAD (... commits, inclusive)`, while JSON still carries the exact endpoint SHA for machine consumers.

Verification:
- `npm run osc -- study --out /tmp/hermes-self-study-inclusive-amend.md` + timestamp-normalized compare — PASS
- `npm test -- --run tests/study.test.ts` — PASS, 16/16
- `npm run build` — PASS
- `npm test -- --run` — PASS, 442/442 before the self-study-only amend
- `./verify.sh --strict` — PASS, 9 pass / 0 fail / 1 existing worktree warning
- `git diff --check` — PASS

**Comment by graphanov** [comment]
@codex review

File: src/study.ts [hunk 7]
```diff
@@ -0,0 +1,460 @@
...
+    );
+  }
+  return headlines;
+}
+
+export function renderStudyMarkdown(report: StudyReport): string {
+  const range = report.commit_range;
+  const rangeText = range.source === 'git' && range.from && range.to
+    ? `${range.from} through HEAD (${range.commit_count ?? 0} commit${range.commit_count === 1 ? '' : 's'}, inclusive)`
+    : 'unavailable (no git history read)';
```
[the remark this answers is not in the document]
Comment (graphanov) [reply src/study.ts]: Fixed in 50e4deb.

Change: valid filtered git ranges with zero commits now render as `0 commits (empty selected git range)` instead of `unavailable`. True unavailable remains reserved for cases where git history could not be read.

Regression: added `renders a valid zero-commit git range as zero rather than unavailable` in `tests/study.test.ts`.

Verification:
- `npm test -- --run tests/study.test.ts` — PASS, 17/17
- `npm run build` — PASS
- `npm test -- --run` — PASS, 443/443 before the self-study-only amend
- `npm run osc -- study --out /tmp/hermes-self-study-zero-range-amend.md` + timestamp-normalized compare — PASS
- `./verify.sh --strict` — PASS, 9 pass / 0 fail / 1 existing worktree warning
- `git diff --check` — PASS

**Comment by graphanov** [comment]
@codex review

**Comment by graphanov** [comment]
@codex review

File: docs/AB_COMPARISON_PROTOCOL.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
+# Controlled Comparison (A/B) Protocol
+
+Status: protocol only — not yet run. Running this comparison is future work, gated on the observational self-study from `.osc/plans/backlog/125-methodology-evidence-harness.md` landing first and on having enough matched tasks/participants to run it credibly. This document is the recipe, not a result.
```
[the remark this answers is not in the document]
Comment (graphanov) [reply docs/AB_COMPARISON_PROTOCOL.md]: Fixed in ff40ff7.

Change: updated the stale plan-125 references in both `docs/AB_COMPARISON_PROTOCOL.md` and `docs/decisions/2026-05-28-pursue-methodology-evidence-harness.md` from `.osc/plans/backlog/125...` to `.osc/plans/done/125...`.

Verification:
- `./verify.sh --strict` — PASS, 9 pass / 0 fail / 1 existing worktree warning
- `npm run osc -- study --out /tmp/hermes-self-study-path-amend.md` + timestamp-normalized compare — PASS
- `git diff --check` — PASS

**Comment by graphanov** [comment]
@codex review
